ssz/cache/utils.py

```diff
@@ -27,7 +27,7 @@ def get_key(sedes: TSedes, value: Any) -> str:
     else:
         # If the serialized result is empty, use sedes name as the key
         if hasattr(sedes, 'element_sedes'):
-            return sedes_name + str(sedes.max_length)
+            return f"{sedes_name}[{sedes.max_length}]({type(sedes.element_sedes).__name__})"
         else:
```

Comment:
we are assuming this is a `List` sedes w/ the access to the `max_length` property -- will all possible structures that bring us here have this property?

Reply:
1. Because `ssz/cache/utils.py` is imported by `ssz/sedes/basic.py`, so it's hacky using `hasattr(sedes, 'element_sedes')` to filter the `List` and `Vector` sedes here. Also, it could be refactored with abstract class: https://github.com/ethereum/py-ssz/pull/88/files#diff-09e4ed3c000952c6e921a58c5c154be5R187https://github.com/ethereum/py-ssz/pull/88/files#diff-09e4ed3c000952c6e921a58c5c154be5R187.
2. Both `Vector` and `List` have `max_length`.